### PR TITLE
CppCheck OutputReportTabular

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes programming style
 - Use CppCheck as guide

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/OutsideEnergySources.cc [hunk 1]
```diff
@@ -175,7 +175,7 @@ void GetOutsideEnergySourcesInput(EnergyPlusData &state)
             EnergyType = DataPlant::PlantEquipmentType::PurchHotWater;
             heatWaterIndex++;
             thisIndex = heatWaterIndex;
-        } else if (EnergySourceNum > NumDistrictUnitsHeatWater && EnergySourceNum <= NumDistrictUnitsHeatWater + NumDistrictUnitsCool) {
+        } else if (EnergySourceNum <= NumDistrictUnitsHeatWater + NumDistrictUnitsCool) {
```
Comment (rraustad) [inline src/EnergyPlus/OutsideEnergySources.cc:178]: [src/EnergyPlus/OutsideEnergySources.cc:178]:(style),[knownConditionTrueFalse],Condition 'EnergySourceNum>NumDistrictUnitsHeatWater' is always true

File: src/EnergyPlus/PhaseChangeModeling/HysteresisModel.cc [hunk 2]
```diff
@@ -95,12 +95,12 @@ namespace Material {
             Tau2 = this->deltaTempMeltingHigh;
             if (updatedTempTDT < TempLowPCM) {
                 phaseChangeState = Phase::Crystallized;
-            } else if (updatedTempTDT >= TempLowPCM && updatedTempTDT <= TempHighPCM) {
+            } else if (updatedTempTDT <= TempHighPCM) {
```
Comment (rraustad) [inline src/EnergyPlus/PhaseChangeModeling/HysteresisModel.cc:98]: [src/EnergyPlus/PhaseChangeModeling/HysteresisModel.cc:98]:(style),[knownConditionTrueFalse],Condition 'updatedTempTDT>=TempLowPCM' is always true

File: src/EnergyPlus/PhotovoltaicThermalCollectors.cc [hunk 3]
```diff
@@ -1450,11 +1444,10 @@ namespace PhotovoltaicThermalCollectors {
...
             k_air = 0.000000000015207 * std::pow(t_film + 273.15, 3.0) - 0.000000048574 * std::pow(t_film + 273.15, 2.0) +
-                    0.00010184 * (t_film + 273.15) - 0.00039333;                        // Dumas, A., and Trancossi, M., SAE Technical Papers, 2009
-            prandtl_air = 0.680 + 0.000000469 * std::pow(t_film + 273.15 - 540.0, 2.0); // The Schock Absorber Handbook, 2nd Ed. John C. Dixon 2007
-            density_air = 101.3 / (0.287 * (t_film + 273.15));                          // Ideal gas law
-            diffusivity_air = k_air / (cp_amb * density_air);                           // definition
-            kin_viscosity_air = mu_air / density_air;                                   // definition
+                    0.00010184 * (t_film + 273.15) - 0.00039333; // Dumas, A., and Trancossi, M., SAE Technical Papers, 2009
+            density_air = 101.3 / (0.287 * (t_film + 273.15));   // Ideal gas law
+            diffusivity_air = k_air / (cp_amb * density_air);    // definition
+            kin_viscosity_air = mu_air / density_air;            // definition
```
Comment (rraustad) [inline src/EnergyPlus/PhotovoltaicThermalCollectors.cc:1450]: [src/EnergyPlus/PhotovoltaicThermalCollectors.cc:1530]:(style),[redundantAssignment],Variable 'prandtl_air' is reassigned a value before the old one has been used.

File: src/EnergyPlus/Photovoltaics.cc [hunk 4]
```diff
@@ -1497,8 +1497,8 @@ namespace Photovoltaics {
                 Real64 &XX,
                 std::function<Real64(EnergyPlusData &state, Real64 const, Real64 const, Real64 const, Real64 const, Real64 const, Real64 const)> FXX,
                 std::function<Real64(EnergyPlusData &state, Real64 const, Real64 const, Real64 const, Real64 const, Real64 const)> DER,
-                Real64 const II, // Autodesk Aliased to XX in some calls
-                Real64 const VV, // Autodesk Aliased to XX in some calls
+                Real64 const &II, // Autodesk Aliased to XX in some calls
+                Real64 const &VV, // Autodesk Aliased to XX in some calls
```
Comment (rraustad) [inline src/EnergyPlus/Photovoltaics.cc]: Debugger showed an uninitialized argument in the call to the NEWTON function for argument II. Changing the argument declarations back to what it was previously corrected the example file failures for GeneratorswithPV and ShopWithBIPVT.

**Comment by nrel-bot-2c** [comment]
@rraustad @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by Myoldmopar** [comment]
Pulled develop in here and it's all happy.  Merging this.  Thanks @rraustad

**Pull request merged**